packages/react-components/src/components/SendBox.tsx

```diff
@@ -217,12 +242,36 @@ export const SendBox = (props: SendBoxProps): JSX.Element => {
   };
 
   /* @conditional-compile-remove(file-sharing) */
-  const renderFileUploadErrorMessage: JSX.Element = useMemo(() => {
+  const uploadInProgressErrorBar = useMemo(() => {
+    const renderError = !!(
+      props.activeFileUploads?.length &&
+      !props.activeFileUploads
+        .filter((fileUpload) => !fileUpload.errorMessage)
+        .every((fileUpload) => fileUpload.isUploaded)
+    );
+    return renderError && <SendBoxErrorBar message={strings.fileUploadsPendingError} />;
+  }, [props.activeFileUploads, strings.fileUploadsPendingError]);
+
+  /* @conditional-compile-remove(file-sharing) */
+  const fileUploadErrorBar = useMemo(() => {
     const fileUploads: ActiveFileUpload[] = props.activeFileUploads || [];
     const latestError = fileUploads.filter((fileUpload) => fileUpload.errorMessage).pop();
-    return <SendBoxErrorBar message={latestError?.errorMessage} timeout={10 * 1000} />;
+    return latestError && <SendBoxErrorBar message={latestError.errorMessage} timeoutDelay={10 * 1000} />;
   }, [props.activeFileUploads]);
 
+  /* @conditional-compile-remove(file-sharing) */
+  const renderSendBoxError = () => {
+    if (showFileUploadsPendingError) {
+      fileUploadsPendingErrorTimeoutRef.current && clearTimeout(fileUploadsPendingErrorTimeoutRef.current);
+      fileUploadsPendingErrorTimeoutRef.current = setTimeout(() => {
+        setShowFileUploadsPendingError(false);
```

Comment:
If this fileUploadsPendingErrorTimeoutRef is just because we want to set some state outside - then we can reuse timeoutDelay logic in SendBoxErrorBar:
1. expose an onDismiss callback in SendBoxErrorBar props, 
2. pass setShowFileUploadsPendingError to it, so it gets called when the errorBar gets dismissed

At this we could save one more timeout object from our component (the less timeout logic we have, the better)

Reply:
Love that idea. Let me try that.